mono/metadata/w32process-unix-bsd.c

```diff
@@ -93,7 +93,29 @@ mono_w32process_get_name (pid_t pid)
 gchar*
 mono_w32process_get_path (pid_t pid)
 {
+#if defined (__OpenBSD__)
+	// No KERN_PROC_PATHNAME on OpenBSD
 	return mono_w32process_get_name (pid);
+#else
+	gsize path_len = PATH_MAX + 1;
+	gchar path [PATH_MAX + 1];
+	gint mib [3];
+#if defined (__NetBSD__)
+	mib [0] = KERN_PROC_ARGS;
```

Comment:
`mib [0]` should be `CTL_KERN` for both NetBSD and FreeBSD/DragonFly.

Reply:
Does that need to bump the MIB further?

Reply:
like?:

```c
        gint mib [4];
        mib [0] = CTL_KERN;
#if defined (__NetBSD__)
        mib [1] = KERN_PROC_ARGS;
        mib [2] = pid;
        mib [3] = KERN_PROC_PATHNAME;
#else // FreeBSD
        mib [1] = KERN_PROC;
        mib [2] = KERN_PROC_PATHNAME;
        mib [3] = pid;
#endif
        if (sysctl (mib, 4, path, &path_len, NULL, 0) < 0) {
```